waku/node/peer_manager/peer_manager.nim

```diff
@@ -619,30 +606,59 @@ proc getNumStreams*(pm: PeerManager, protocol: string): (int, int) =
   return (numStreamsIn, numStreamsOut)
 
 proc pruneInRelayConns(pm: PeerManager, amount: int) {.async.} =
-  let (inRelayPeers, outRelayPeers) = pm.connectedPeers(WakuRelayCodec)
+  let (inRelayPeers, _) = pm.connectedPeers(WakuRelayCodec)
   let connsToPrune = min(amount, inRelayPeers.len)
 
   for p in inRelayPeers[0..<connsToPrune]:
     asyncSpawn(pm.switch.disconnect(p))
 
 proc connectToRelayPeers*(pm: PeerManager) {.async.} =
-  let (inRelayPeers, outRelayPeers) = pm.connectedPeers(WakuRelayCodec)
-  let maxConnections = pm.switch.connManager.inSema.size
-  let totalRelayPeers = inRelayPeers.len + outRelayPeers.len
-  let inPeersTarget = maxConnections - pm.outRelayPeersTarget
+  var peersToConnect: seq[RemotePeerInfo]
+  var peersToDisconnect: int
+
+  for shard in pm.wakuMetadata.shards.items:
+    let peers = pm.peerStore.getPeersByShard(uint16(pm.wakuMetadata.clusterId), uint16(shard))
+
+    let connectedInPeers = peers.filterIt(
```

Comment:
main reason for using `pm.connectedPeers(WakuRelayCodec)` is that it returns the peers connected with that codec.

in other words, `.isConnected(it.peerId)` might give you peers that are connected but with other protocol that is not relay, and afaik here we need to count relay peers.

Reply:
Ah good catch!

Reply:
@SionoiS cant see it fixed in the latest commit?

Reply:
@alrevuelta  L#622 the peers are filtered by protocol (WakuRelayCodec)